chalice/config.py

```diff
@@ -151,16 +151,13 @@ def lambda_python_version(self) -> str:
         if major == 2:
             return 'python2.7'
         # Python 3 for backwards compatibility needs to select python3.6
-        # for python versions 3.0-3.6. 3.7 and higher will use python3.7.
+        # for python versions 3.0-3.6. 3.7-3.10 will use their version.
+        # 3.11 and higher will use 3.11
         elif (major, minor) <= (3, 6):
             return 'python3.6'
-        elif (major, minor) <= (3, 7):
-            return 'python3.7'
-        elif (major, minor) <= (3, 8):
-            return 'python3.8'
-        elif (major, minor) <= (3, 9):
-            return 'python3.9'
-        return 'python3.10'
+        elif (major, minor) <= (3, 10):
```

Comment:
Originally I just added an additional branch, but pylint did not like having this many return statements.